AudioPlayer/AudioPlayer/player/AudioPlayerState.swift

```diff
@@ -25,13 +25,13 @@ public enum AudioPlayerError: Error {
 /// - stopped: The player is stopped.
 /// - waitingForConnection: The player is waiting for internet connection.
 /// - failed: An error occured. It contains AVPlayer's error if any.
-public enum AudioPlayerState {
-    case buffering
-    case playing
-    case paused
-    case stopped
-    case waitingForConnection
-    case failed(AudioPlayerError)
+@objc public enum AudioPlayerState: Int {
```

Comment:
Right. The advantage of this is that there's less state to handle (i.e. set `failedError` to nil at some point and also exposing it to objc - right now I don't think it's visible in objc but correct me if I'm wrong) but yeah I didn't really think it through for the protocol.
Anyway, how about wrapping the state into a class then? I'm not a huge fan of this solution but I'd prefer the state to contain the error instead of the player. What are you thoughts on this?